src/utils.jl

```diff
@@ -0,0 +1,6 @@
+function sum_if_not_none_or_initialize(A,B)
+    if isnothing(A) 
+        return deepcopy(copy((B)))
+    end
+    return A + B
+end
```

Comment:
I think I have used copy() in vain, but the function is executed in loop, so I will need a deepcopy of B when A is nothing (otherwise B points to A).

Reply:
> so I will need a deepcopy of B when A is nothing (otherwise B points to A).

What does this mean?

Reply:
Basically it is a recursive copy of objects, but here it is just to ensure that we don't pass a reference of A to B. Shallow copy could be used, but I though there must be some motive in using deepcopy in python too (like maybe variance could be an array in some case so shallow copy doesn't guarantee that A and B are independent).

Reply:
Specifically I'm missing what

> (otherwise B points to A)

means. Maybe a concrete reproducible example would help.

Reply:
The specific use-case is here: https://github.com/StingraySoftware/stingray/blob/1bd574a284dfe60177e3313d50cc5700f4d1ec42/stingray/fourier.py#L1005-L1007
As you can see if I just use `sum_if_not_none_or_initialize(::Nothing,B) = B` , it will point to the reference of B. Ex:
<img width="379" alt="image" src="https://user-images.githubusercontent.com/77021852/175765891-4ca9ce8e-8b8d-4e5d-95bb-45358ea8cf8c.png">

Reply:
As a general remark, never show screenshots, whoever wants to help you can't copy from that, moreover if someone wants to come back to this discussion whatever is written in the screenshot doesn't show up in search. They aren't really much useful.

That said, I'm not sure that's a problem in practice here.

Also, I think this discussion is a bit out of the point now: I believe you should first understand how to avoid the overuse of nothing and write more idiomatic Julia code here, at which point a new solution may arise.

Reply:
Hi @giordano, @Aman-Pandey-afk  : 
This is the use case: in another function, I am calculating (e.g.) an average of many periodograms produced over a number of short time series. In each loop, I produce a periodogram that has to be summed to the total periodogram. The summed periodogram is initialized as `None`. In the first loop, this function will find that the total periodogram is None, and initialize it to a deep copy of the first periodogram. From that point on, all subsequent arrays will be summed to the total periodogram in place.
The reason for the deep copy is to avoid side effects on the first periodogram, in case it is useful later.

Reply:
Ok, but the `deepcopy` is a red herring here, my main point is that we should move away from overusing `nothing`.  There _are_ definitely good use cases for `nothing` in Julia, but I feel like all the uses here come from the fact it was used (for good reasons!) in Python, not that it's necessary in Julia.